plugins/korean/Agitoon.ts

```diff
@@ -8,14 +8,20 @@ class Agitoon implements Plugin.PluginBase {
   id = 'agit.xyz';
   name = 'Agitoon';
   icon = 'src/kr/agitoon/icon.png';
-  site = 'https://agit655.xyz';
-  version = '2.0.1';
+  site = 'https://agit657.xyz';
+  version = '3.0.0';
+  static url: string | undefined;
 
   async popularNovels(
     pageNo: number,
     { filters, showLatestNovels }: Plugin.PopularNovelsOptions,
   ): Promise<Plugin.NovelItem[]> {
-    const res = await fetchApi(this.site + '/novel/index.update.php', {
+    if (!Agitoon.url) {
+      Agitoon.url =
+        (await fetchApi(this.site).then(res => res.url.replace(/\/$/, ''))) ||
+        this.site;
+    }
+    const res = await fetchApi(Agitoon.url + '/novel/index.update.php', {
```

Comment:
I think you should call this in constructor instead.
And static keyword is not necessary tho .

Reply:
Wouldn't that make a request when the app is opened?

Reply:
Im going to implement lazy initialization for plugins. so nyo